datafusion/sqllogictest/test_files/union.slt

```diff
@@ -466,8 +466,8 @@ UnionExec
 ----AggregateExec: mode=FinalPartitioned, gby=[Int64(1)@0 as Int64(1)], aggr=[]
 ------CoalesceBatchesExec: target_batch_size=8192
 --------RepartitionExec: partitioning=Hash([Int64(1)@0], 4), input_partitions=4
-----------RepartitionExec: partitioning=RoundRobinBatch(4), input_partitions=1
-------------AggregateExec: mode=Partial, gby=[1 as Int64(1)], aggr=[]
+----------AggregateExec: mode=Partial, gby=[1 as Int64(1)], aggr=[]
```

Comment:
this seems like a better plan as it will repartition the input (though it probably doesn't matter as the input produces a single row)

Reply:
Maybe unnecessary parallelism could be improved with a simple statistics-based rule: if the input size is less than the batch size, nothing could be gained from the increased parallelism.